submissions/chris-voitova/friends-app/script.js

```diff
@@ -0,0 +1,199 @@
+const url = "https://randomuser.me/api/?results=16";
+const usersContainer = document.querySelector(".users-container");
+
+const initState = {
+  maleGender: false,
+  femaleGender: false,
+  allGenders: true,
+};
+
+let state = Object.assign({}, initState);
+
+const errorMessage = (message) => `<p class="error-message">${message}</p>`;
+
+const userTemplate = (photo, name, email, gender, age) => {
+  return `
+   <div class="user">
+      <img
+         class="user__photo"
+         src=${photo}
+         alt="user-photo"
+      />
+      <div class="user__info">
+         <h3 class="user__name">${name}</h3>
+         <a class="user__email" href="mailto:${email}">
+           ${email}
+         </a>
+         <span class="user__gender">gender: ${gender}</span>
+         <span class="user__age">age: ${age}</span>
+      </div>
+   </div>`;
+};
+
+async function getUsers() {
+  try {
+    const response = await fetch(url);
+    const json = await response.json();
+    return json.results;
+  } catch {
+    usersContainer.innerHTML += errorMessage("oops :( users fetch failed");
+  }
+}
+
+async function app() {
+  const initUsers = await getUsers();
+  const users = initUsers.slice();
+  renderUsers(users);
+  handleSearch(users);
+  handleFilterByGender(users, initUsers);
+  handleSort(users, "#age-sort", sortByAge, [
+    "#sort-ascending",
+    "#sort-descending",
+  ]);
+  handleSort(users, "#name-sort", sortByName, [
+    "#sort-az",
+    "#sort-za"
+   ]);
+  handleResetToDefault(initUsers);
+}
+
+const renderUsers = (users) => {
+  let renderedUsers = "";
+  users.forEach(
+    ({
+      picture: { medium: photo },
+      name: { first, last },
+      email,
+      gender,
+      dob: { age },
+    }) => {
+      const name = `${first} ${last}`;
+      renderedUsers += userTemplate(photo, name, email, gender, age);
+    }
+  );
+  usersContainer.innerHTML = "";
+  usersContainer.innerHTML += renderedUsers;
+};
+
+const handleSort = (
+  users,
+  container,
+  sortFunc,
+  [ascendingTrigger, descendingTrigger]
+) => {
+  container = document.querySelector(container);
+  ascendingTrigger = document.querySelector(ascendingTrigger);
+  descendingTrigger = document.querySelector(descendingTrigger);
+
+  container.addEventListener("click", ({ target }) => {
+    const filteredUsers = checkSelectedState(users);
+
+    if (target === ascendingTrigger) {
+      sortFunc(filteredUsers);
+      renderUsers(filteredUsers);
+    }
+    if (target === descendingTrigger) {
+      sortFunc(filteredUsers).reverse();
+      renderUsers(filteredUsers);
+    }
+  });
+};
+
+const checkSelectedState = (users) => {
+  let data;
+  state.maleGender
```

Comment:
Nested ternary operators are bad in 99,9% cases - they hard to read and support. Also, they not provide any additional features.